modules/qc.module.nf

```diff
@@ -130,12 +130,29 @@ process aggregate_kmer {
 }
 
 
+process validatebam {
+
+    tag {"${grouping} -> ${name}" }
+
+    label 'sm'
+
+    input:
+        tuple grouping, name, path("in.bam"), file("in.bam.bai")
+    output:
+        tuple grouping, path("${name}.validatesamfile.txt")
+
+    """
+        picard ValidateSamFile I=in.bam MODE=SUMMARY > ${name}.validatesamfile.txt
```

Comment:
As I mentioned before, I think `samtools quickcheck` is good enough here.

Reply:
I still couldn't figure out how the corruption happened, so it makes me feel a bit more reassured to check the whole file, and not to find out about corrupted bam file in the middle of variant calling. But I will keep it in mind that there is simpler and faster check! Thanks!

Reply:
I'll leave it up to you which way to keep it. But `samtools quickcheck` will address your concerns regarding a bam becoming corrupted midway through calling.

samtools quickcheck:

> Quickly check that input files appear to be intact. Checks that beginning of the file contains a valid header (all formats) containing at least one target sequence and then seeks to the end of the file and checks that an end-of-file (EOF) is present and intact (BAM and CRAM only).

If the bam is corrupted in the middle, `samtools quickcheck` will flag it as such because it will not have the end of file marker.